compiler/qsc_eval/src/lib.rs

```diff
@@ -51,6 +51,13 @@ pub enum Error {
     #[diagnostic(code("Qsc.Eval.DivZero"))]
     DivZero(#[label("cannot divide by zero")] PackageSpan),
 
+    #[error("cannot record output for empty array")]
+    #[diagnostic(help(
+        "returning an empty array in the final result of a program is not supported"
+    ))]
+    #[diagnostic(code("Qsc.Eval.EmptyArrayOutput"))]
+    EmptyArrayOutput(PackageSpan),
```

Comment:
Nit: I don't know if this would count as way too much work just to add one new error, but I think it would be nice if this was under a different error enum specific to QIR generation.

Reply:
Yeah, I considered that, and I think you are right. It sort of belongs as a QIR-only error, and then QIR generation would return an aggregate error that is either evaluation error or something QIR specific. That would require updating all the callers of QIR generation, which definitely makes the change bigger but might be worth it, especially if we expect more QIR codegen specific errors later (and I do).